src/app/games/yinsh/Yinsh.component.ts

```diff
@@ -0,0 +1,326 @@
+import { Component } from '@angular/core';
+import { HexagonalGameComponent } from 'src/app/components/game-components/abstract-game-component/HexagonalGameComponent';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MoveEncoder } from 'src/app/jscaip/Encoder';
+import { HexaLayout } from 'src/app/jscaip/HexaLayout';
+import { FlatHexaOrientation } from 'src/app/jscaip/HexaOrientation';
+import { MessageDisplayer } from 'src/app/services/message-displayer/MessageDisplayer';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPValidation } from 'src/app/utils/MGPValidation';
+import { YinshFailure } from './YinshFailure';
+import { YinshGameState } from './YinshGameState';
+import { YinshLegalityStatus } from './YinshLegalityStatus';
+import { YinshMinimax } from './YinshMinimax';
+import { YinshCapture, YinshMove } from './YinshMove';
+import { YinshPiece } from './YinshPiece';
+import { YinshRules } from './YinshRules';
+
+interface CaseInfo {
+    coord: Coord,
+    coordinates: string,
+    caseClasses: string[],
+    pieceClasses: string[],
+    isPiece: boolean,
+    center: Coord,
+}
+
+interface ViewInfo {
+    caseInfo: CaseInfo[][],
+    possibleCaptures: YinshCapture[],
+    selectableRings: Coord[],
+    pieceSize: number,
+    sideRings: [number[], number[]], // for each player: array of [1, 2, 3] means 3 rings to show
```

Comment:
suboptimal way of encoding 1 2 or 3 :p

Reply:
oui mais pour looper dessus dans le html je ne connais pas d'autre moyen ! (pas moyen de faire un "for i from 0 to 3", du coup c'est "for i in sideRings[0]") Une autre idée ?

Reply:
Ah oui vu comme ça, oui non faudra soit garder ça soit un subterfuge moyen parmis:
https://stackoverflow.com/questions/36535629/repeat-html-element-multiple-times-using-ngfor-based-on-a-number
Le pipe du début étant pas dégueux :/

Reply:
ou plutôt le [].constructor(n) qui requiert aucun code supplémentaire à maintenir ?

Reply:
Oui c'est léger, faisons ça!